### PR TITLE
Load path pedestrian focus styles from comparison summaries

### PR DESCRIPTION
**Pull request by ebelo**
## Summary

- let the path/pedestrian focus report load QGIS-preprocessed style paths from an all-camera comparison `summary.json`
- derive the trusted comparison debug root from `.../all-cameras/summary.json`, while still allowing an explicit trusted root in tests
- resolve each camera manifest and extracted style path under that trusted root before loading styles
- surface path/pedestrian line and fill color counts plus compact color samples alongside width/dash/opacity controls
- keep explicit `--qgis-style-json CAMERA=PATH` overrides for manual runs
- add tests for comparison-summary discovery, trusted-path rejection, CLI summary loading, and color samples

## Validation

- `python3 -m pytest tests/test_mapbox_outdoors_path_pedestrian_focus.py -q --tb=short` (`17 passed`)
- `python3 -m py_compile validation/mapbox_outdoors_path_pedestrian_focus.py tests/test_mapbox_outdoors_path_pedestrian_focus.py`
- `git diff --check`
- `python3 -m pytest tests/ -x -q --tb=short` (`2265 passed, 173 skipped, 162 subtests passed`)
- `python3 validation/mapbox_outdoors_path_pedestrian_focus.py --road-features-json debug/mapbox-outdoors-road-features/all-cameras/20260520T010913Z/road-features.json --comparison-summary-json debug/mapbox-outdoors-comparison/all-cameras/20260520T003516Z/summary.json`

Latest local artifact: `debug/mapbox-outdoors-path-pedestrian-focus/20260520T020144Z/summary.md`

## Issue

Part of #949.

### CODE REVIEW

**Comment by ebelo** [comment]
Addressed Greptile follow-up in b507a62: added a regression for a trusted manifest pointing at an untrusted `qgis_preprocessed_style` output, and updated the trusted-root error to include the resolved offending path. Re-ran focused tests (`16 passed`), py_compile, diff check, full suite (`2264 passed, 173 skipped, 162 subtests passed`), and regenerated `debug/mapbox-outdoors-path-pedestrian-focus/20260520T015959Z/summary.md`.

File: validation/mapbox_outdoors_path_pedestrian_focus.py [hunk 1]
```diff
@@ -70,6 +71,53 @@ def load_json_object(path: Path) -> dict[str, object]:
...
+    return resolved
+
+
+def qgis_style_paths_from_comparison_summary(
+    comparison_summary: Mapping[str, object],
+    *,
+    summary_path: Path | None = None,
+    trusted_root: Path | None = None,
+) -> dict[str, Path]:
+    root = DEFAULT_COMPARISON_OUTPUT_ROOT if trusted_root is None else trusted_root
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_path_pedestrian_focus.py]: Addressed in fe35a63: comparison-summary discovery now derives the trusted root from the supplied summary path by locating the `all-cameras` directory, so custom comparison output roots work without monkey-patching.

File: tests/test_mapbox_outdoors_path_pedestrian_focus.py [hunk 2]
```diff
@@ -115,6 +120,53 @@ def test_load_json_object_requires_object(self):
...
+                    }
+                ]
+            }
+
+            with self.assertRaises(ValueError):
+                qgis_style_paths_from_comparison_summary(
+                    comparison_summary,
+                    summary_path=Path(tmpdir) / "summary.json",
+                    trusted_root=Path(tmpdir),
+                )
```
[the remark this answers is not in the document]
Comment (ebelo) [reply tests/test_mapbox_outdoors_path_pedestrian_focus.py:168]: Addressed in fe35a63: added a regression test covering a trusted manifest whose `qgis_preprocessed_style` output points outside the derived comparison root.

File: validation/mapbox_outdoors_path_pedestrian_focus.py [hunk 3]
```diff
@@ -70,6 +71,53 @@ def load_json_object(path: Path) -> dict[str, object]:
     return loaded
 
 
+def _resolve_trusted_debug_path(raw_path: str, *, base_dir: Path, trusted_root: Path) -> Path:
+    path = Path(raw_path)
+    candidate = path if path.is_absolute() else base_dir / path
+    resolved = candidate.resolve()
+    if not resolved.is_relative_to(trusted_root.resolve()):
+        raise ValueError(f"Comparison artifact path must stay under {trusted_root}")
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_path_pedestrian_focus.py]: Addressed in fe35a63: the trusted-path error now includes the rejected resolved path plus the required root, so parser errors identify the offending manifest or style path.

**Comment by ebelo** [comment]
Additional Greptile follow-up in 4f6458e: the CLI path now derives the trusted comparison root from `.../all-cameras/summary.json` instead of relying on the default root when a concrete summary path is provided. I also kept the explicit-root path covered separately by renaming the duplicate untrusted-style test so both trust-boundary cases run. Re-ran focused tests (`17 passed`), py_compile, diff check, full suite (`2265 passed, 173 skipped, 162 subtests passed`), and regenerated `debug/mapbox-outdoors-path-pedestrian-focus/20260520T020144Z/summary.md`.
